### ISSUE TITLE
[feat] Blobstream Contracts Integration Tests

### ISSUE THREAD

**Issue by b-j-roberts** [keep-starknet-strange/blobstream-starknet#85]
Create integration tests (scripts) for the Blobstream contract(s) by doing the following :
- Launch a `Katana` node
- Declare & Deploy the `BlobstreamX` contract ( & dependencies such as the gateway ) to the `Katana` node
- Make calls to public functions on `BlobstreamX` contract & check state updates are correct

Possibly use mock implementations where needed.

#### References

https://github.com/keep-starknet-strange/blobstream-starknet/blob/main/src/lib.cairo

**Comment by b-j-roberts** [comment]
This is currently blocked. Waiting for `Katana` to upgrade the supported JSON-RPC spec to 0.6.0

It was posted that this should be soon https://t.me/dojoengine/5922

**Comment by b-j-roberts** [comment]
The decision to switch from `Katana` to `starknet-devnet-rs` has unblocked this. `Katana` does now support 0.6.0, but this decision was made prior to that. I think this is still the best way to go because forking `starknet-devnet-rs` doesn't require forking the entire dojo engine. ( Forking will be required to unblock Groth EC pairings with new syscalls for now )

**Comment by b-j-roberts** [comment]
#### Update

We have decided to integrate with Herodotus storage proof's / API. So, instead of using a devnet for integration tests, we will need to do tests on Sepolia for the integration.
This would include setting up deployment and upgrading scripts for the `BlobstreamX` contracts on `Sepolia`, script for sending a blob to Celestia, using the request data commitments Herodotus script, and verifying DA for the blob on Starknet through `BlobstreamX`.

#### References
- [Herodotus integration](https://github.com/keep-starknet-strange/blobstream-starknet/issues/124)
- [Herodotus data commitments request script](https://github.com/keep-starknet-strange/blobstream-starknet/pull/120)
- [Posting blobs to Celestia](https://docs.celestia.org/developers/node-tutorial)

**Comment by thomas192** [comment]
So remaining scripts to do are : 
- updating Starknet `BlobstreamX` data commitments from the Herodotus Fact registry
- verifying DA for the blob on Starknet through `BlobstreamX`

Can I work on these ? If so, is `starkli` the way to go ?

**Comment by drspacemn** [comment]
@thomas192 yes this is a good summary of the remaining scripts. Can you take #138 and I'll take #139 ?

**Comment by gogo2464** [comment]
duplicate of https://github.com/keep-starknet-strange/blobstream-starknet/issues/136 ?
